metric-anomaly-detector/src/main/java/org/hypertrace/alert/engine/metric/anomaly/detector/MetricQueryBuilder.java

```diff
@@ -267,18 +307,18 @@ static Expression createStringLiteralExpression(String value) {
   }
 
   String getTimestampAttributeId(String tenantId, String attributeScope) {
-    Iterator<AttributeMetadata> attributeMetadataIterator =
-        attributesServiceClient.findAttributes(
-            Map.of(AlertRuleEvaluator.TENANT_ID_KEY, tenantId),
-            AttributeMetadataFilter.newBuilder().addScopeString(attributeScope).build());
-
-    while (attributeMetadataIterator.hasNext()) {
-      AttributeMetadata metadata = attributeMetadataIterator.next();
-      if (metadata.getKey().equals(START_TIME_ATTRIBUTE_KEY)) {
-        return metadata.getId();
-      }
+    try {
+      return attributeServiceCache.get(Pair.of(tenantId, attributeScope));
+    } catch (ExecutionException e) {
+      LOG.error(
```

Comment:
So, this will log failure in executing task, right?

Reply:
try/catch block was required for the cache